### PR TITLE
pdf_new_name

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: dpx/src/dpx_pdfobj.rs [hunk 1]
```diff
@@ -1156,31 +1129,40 @@ pub unsafe extern "C" fn pdf_set_string(
     };
 }
 /* Name does *not* include the /. */
-#[no_mangle]
-pub unsafe extern "C" fn pdf_new_name(mut name: *const i8) -> *mut pdf_obj {
+pub unsafe fn pdf_new_name(name: &str) -> *mut pdf_obj {
```
Comment (burrbull) [inline dpx/src/dpx_pdfobj.rs:1132]: `pdf_new_name` splitted on `pdf_new_name(name: &str)` and `pdf_copy_name(name: *const i8)`.
Use Rust allocator.

**Comment by cormacrelf** [comment]
Generally pdfobj stuff would be superseded by #73

**Comment by cormacrelf** [comment]
This moves call sites in the right direction so good, but probably don’t go too far down the pdfobj rabbit hole

**Comment by burrbull** [comment]
> rabbit hole

That's why I I do it by small steps.

**Comment by crlf0710** [comment]
Seems good, merging~

**Pull request merged**